### ISSUE TITLE
Update beanshell.org

### ISSUE THREAD

**Issue by nickl-** [beanshell/beanshell#82]
Created on behalf of @pgiffuni

See if we can update beanshell.org so that it point to github and update the license information there.

**Comment by nickl-** [comment]
This also relates to #17 Change groupId to org.beanshell

**Comment by pgiffuni** [comment]
I think this can only be done by Pat or someone lese that has access to the website. Updating thesourceforge site is also somethng that should be done.

**Comment by nickl-** [comment]
@pgiffuni with regards to SourceForge project site you are referring to open ticket #44 

@patniemeyer  made an appearance requesting I email them to get access but I still haven't received any response to my numerous pings.

**Comment by omarpta** [comment]
How can i find the latest .jar version of the latest sources without have to clone and build the project?
thanks in advance

**Comment by nickl-** [comment]
@omarpta There is more information at #81 about the next release and I also added a [road map](https://github.com/beanshell/beanshell#development-road-map) on the project description to show where we re at.

We are not quite ready for a release yet, the only way to get the latest source is to clone the merge-fork-beanshell2 branch and build with `mvn install`

We could really use your help if you are also eager for the release, please see #72

**Comment by pgiffuni** [comment]
This is done at a basic level now.

**Comment by nickl-** [comment]
I guess we could include a built jar file as part of the project sources, which we can update sporadically. It will not suffice as a release but circumvents the requirement to build the project.

This seems to a practice with other projects but I can imagine getting inundated with requests to rebuild the jar file. I would much rather address user issues with building the project or reports of failing unit tests, these goes towards improved health. Much preferred to spoon feeding which has no benefit.

Perhaps I'm being harsh but we need all the help we can get.

**Comment by nickl-** [comment]
@ebourg made a valid comment on the web site

> Who controls the beanshell.org domain? I think it's important to clarify the situation and explain that this project isn't a random fork but the official continuation of the old beanshell code. The domain is going to expire in 44 days, if it isn't renewed http://beanshell.github.io should be used to advertise the project.

**Comment by nickl-** [comment]
#208 reports applets not working

http://www.beanshell.org/manual/appletmode.html

**Comment by nickl-** [comment]
@genman gives constructive criticism at #102 

> A project like yours needs (_deserves_) a better web site, not something that looks like something authored from MS Frontpage 95.

**Comment by nickl-** [comment]
@erh submitter a patch for the manual #475

**Comment by pgiffuni** [comment]
I can't find servetmode.xml in the website:
~/www/beanshell % find ./ -name "servletmode.xml" -print

The manual is in html and pdf only :-/.
I am slowly improving the html layout, but it may take time.
I'll see what I can do about the other issues.

**Comment by pgiffuni** [comment]
In an attempt to provide some consistent updates, I did some changes to the website, however there is a lot of work to be made still . We should upload the website into github to make it easier to edit it and it may be a good idea to consider a re-design at some point.

It would also be nice to have the old releases linked on github so that we don't carry the old tarballs in the website anymore, but I guess that would mean merging the old source repos as well.

**Comment by nickl-** [comment]
> I can't find servletmode.xml in the website:

I found the manual sources... the `.xml` in the `beanshell-site.tar.gz` bundle from google drive at folder `beanshell/BeanShell/docs/manual` not sure how current that is.

Should I add this under source control?

**Comment by nickl-** [comment]
> It would also be nice to have the old releases linked on github so that we don't carry the old tarballs in the website anymore, but I guess that would mean merging the old source repos as well.

We could just convert them to git format and keep them as legacy repos, without trying to merge. I am thinking separate projects leaving the beanshell/beanshell project unchanged. Will start looking into this...

**Comment by pgiffuni** [comment]
> I found the manual sources... the .xml in the beanshell-site.tar.gz bundle from google drive at folder beanshell/BeanShell/docs/manual not sure how current that is.
> 
> Should I add this under source control?

Is it the same documentation we have in the Wiki? Yes, we should have it available: I don't have it in my website mirror.

> It would also be nice to have the old releases linked on github so that we don't carry the old tarballs in the website anymore, but I guess that would mean merging the old source repos as well.
> 
> We could just convert them to git format and keep them as legacy repos, without trying to merge. I am thinking separate projects leaving the beanshell/beanshell project unchanged. Will start looking into this...

Convert the to git, then rebase as necessary to have everything into a single tree. That will probably regenerate all the git checksums but that is OK: We need to paste those repos only once, I think.

**Comment by nickl-** [comment]
> Convert them to git as individual repos, then git rebase as necessary to have everything into a single tree. That will probably regenerate all the git checksums but that is OK: We need to paste those repos only once, I think.

Ouch! That doesn't sound like fun =(

How far back do we have history in this tree anyway?

**Comment by nickl-** [comment]
O no that's no good beanshell/beanshell@217779f7aa2ac07de19f2e73ba0fb1e2d90fc0c4 looks like `copy/paste/git init` we will have to rebase you are right.

**Comment by pgiffuni** [comment]
FWIW, and clarifying that I don't know myself around git too well: the apache-extras stuff was all done with Subversion originally and later converted to git just before moving to github. 

My subversion mentality doesn't let me grok the full value of git hashes, so I wont miss having to recalculate new values ;).

**Comment by nickl-** [comment]
The subversion tree was pulled down (file structure only) and a new git repository was created starting from scratch.

What should've been done is the subversion repo was supposed to be converted to git translating all the commits, tags, branches into git equivalents. Apparently this was not done.

The sourceforge repo is CVS... I assume that comes from before the SVN.

I am sure what we gain from recollecting the history will outweigh what we loose with changed hashes.

**Comment by pgiffuni** [comment]
> The sourceforge repo is CVS... I assume that comes from before the SVN.

Yes. FWIW, cvs2svn does the job straightforward:
cvs2svn Statistics:
------------------
Total CVS Files:               550
Total CVS Revisions:          2805
Total CVS Branches:            379
Total CVS Tags:               5183
Total Unique Tags:              18
Total Unique Branches:           1
CVS Repos Size in KB:        13697
Total SVN Commits:             296
First Revision Date:    Thu Mar 30 15:29:30 2000
Last Revision Date:     Tue Aug 24 11:09:01 2004

**Comment by nickl-** [comment]
cvs2svn? I am confused... we need to get it to git. 

Are you planning on rebasing the 2 svn repos first and the convert to git or what is the plan now?

**Comment by pgiffuni** [comment]
cvs2svn claims it can convert to git as well, but even if not, svn is easy to convert to git (with git-svn).

I am currently just playing with things but I discovered that in order to rebase we need to move both repos to git first.

**Comment by nickl-** [comment]
> I am currently just playing with things but I discovered that in order to rebase we need to move both repos to git first.

Makes sense... push them to separate branches and rebase HEAD against the new branch before deleting them. Glad you are running with this I am stuck knee deep in this current patch and haven't hah a chance for much anything =/

**Comment by pgiffuni** [comment]
To be honest , I wasn't planning on doing it b/c I don't know my way around git/github very well.

Rebasing both old repos can only be done in git because they have different origins so in SVN I just can't "replay" one repository on top of the other. I think I can upload both repos without much trouble (the CVS stuff is bigger), getting them in unified form is another matter.

**Comment by nickl-** [comment]
I could remember doing something like a rebase was it integrate damn it's been too long I can't remember anymore will have to rtfm again. Those were definitely from the same repo never had to try and replay two different svn repos on top of each other.

git I use every day and have never not been able to do something I needed to... those kudos go to git though

**Comment by nickl-** [comment]
Typical

> The term “reintegrating” comes from the merge option --reintegrate. This option is deprecated in Subversion 1.8 (which automatically detects when a reintegrate merge is needed), but is required for Subversion 1.5 through 1.7 clients when performing reintegrate merges.

At least I was close =)

Agreed convert both to git and we can take it from there. I can create you two empty branches if that would help?

**Comment by pgiffuni** [comment]
I can manage creating the branches.. I think I could even do the rebase locally in git. before putting it up.

Ugh.. there's no hurry right?  I have a git book here and  I want to spend some time on it (heck it looks fun): but I don't have time over the weekend :(.

**Comment by nickl-** [comment]
No hurry =) joy learning git is too phun!  =)

The way I imagine it would require the following steps.

Create empty branches for cvs and svn.
Push the converted repositories to these empty branches.
Rebase svn over cvs.
Rebase master over svn branch.
Push master to GitHub.

You can do everything offline until pushing master to GitHub. I will then rebase the development branch over master again.

**Comment by pgiffuni** [comment]
FWIW, I started the repo conversion but I am somewhat stuck: it appears I have to setup svnserver in order to convert the svn part of the repo.
For the time being the old (CVS) repo has been converted to git using cvs2git and named [bsh.git.tgz](http://people.apache.org/~pfg/beanshell/).
I also have the SVN dumps for both the CVS (converted) and SVN repos there.

**Comment by pgiffuni** [comment]
OK, I have now done git two repos from thwe SVN repos feel free to get them [here](http://people.apache.org/~pfg/beanshell/).
bsh-old.git.tgz (12M) has the CVS converted repo.
bsh.git.tgz (5.3M) the repo converted from SVN.
My previous attempt to use cvs2git dodn't go well, so both were converted with git-svn which may mean they still have some garbage (although I was able to convert the names rather nicely to the github usernames),

**Comment by nickl-** [comment]
@pgiffuni awesome very glad I didn't have to do this myself. tyvm

Will take a look at what you made and do the necessary to make it so.

Well done!!!

**Comment by pgiffuni** [comment]
I've been ratherr busy lately and couldn't finish: we still have to match the transtions between repositories. Apparently there are changes between the end of one repository and the start of the other. Ditto for the start of the github repo.

**Comment by nickl-** [comment]
@pgiffuni also been preoccupied and haven't had a chance to look at this yet.

> we still have to match the transtions between repositories. Apparently there are changes between the end of one repository and the start of the other

I don't think that should be a problem. Tx for the heads up.

**Comment by nickl-** [comment]
@pgiffuni Merged repos for master and merge-fork-beanshell2 with your converted repositories.

Thinking we should create a new project for website content... your thoughts?

**Comment by nickl-** [comment]
We should also look into this: https://help.github.com/articles/verifying-your-organization-s-domain/

**Comment by pgiffuni** [comment]
Nice job! I was wondering how much of the original code is left but unfortunately the tab -> spaces conversion hides that information. Nevermind, I can live with it ;).

 It would also be nice to have tags for the old releases., I don't want to point to old releases in the website but they are of historical value.

For the website: we can still leave it where it is now and move it later, or perhaps start the new site om github but leave beanshell.org pointing to the old one until its finished. The domain records correspond to Pat but he said he would do it if we wanted that.

For now there's a lot to do: a new website using bootstrap seems the way forward but my hands are busy now.

**Comment by pgiffuni** [comment]
I added some tags corresponding to old releases.
Since some of the tagas are not clear (in partidular 2.0b4), I used some help from the Waysback machine.

**Comment by nickl-** [comment]
+1 on bootstrap for new site.

**Comment by pgiffuni** [comment]
The new website is setup at https://beanshell.github.io/ .
Relocation is pending.

**Comment by nickl-** [comment]
Nice one!!! Yeah! \o/

I think this is the docs on how to point beanshell.org to the new location 
https://help.github.com/en/articles/using-a-custom-domain-with-github-pages

**Comment by nickl-** [comment]
@pgiffuni I noticed a diagram in the docs not showing.

https://beanshell.github.io/manual/classpath.html#Class_Loading_and_Class_Path_Management

Which works on beanshell.org

http://beanshell.org/manual/classpath.html#Class_Loading_and_Class_Path_Management

Haven't looked into it any further so can't say why it is or if there are others, just bringing it to your attention.

**Comment by nickl-** [comment]
@pgiffuni Question: do you know if the documentation xslts are here and what the building procedures for producing the html are?

**Comment by nickl-** [comment]
Closed this has been completed for now.
